extractors/README.md

```diff
@@ -0,0 +1,110 @@
+
+## Audience
+
+**This README is targeted at middleware developers and contributors.** If you are a Fiber framework user looking to use extractors in your application, please refer to the [Extractors Guide](https://docs.gofiber.io/guide/extractors) instead.
+
+## Architecture
+
+### Core Types
+
+- `Extractor`: Core extraction function with metadata
+- `Source`: Enumeration of extraction sources (Header, AuthHeader, Query, Form, Param, Cookie, Custom)
+- `ErrNotFound`: Standardized error for missing values
+
+### Extractor Structure
+
+```go
+type Extractor struct {
+    Extract    func(fiber.Ctx) (string, error)  // Extraction function
+    Key        string                           // Parameter/header name
+    Source     Source                           // Source type for inspection
+    AuthScheme string                           // Auth scheme (FromAuthHeader)
+    Chain      []Extractor                      // Chained extractors
+}
+```
+
+### Available Functions
+
+- `FromAuthHeader(authScheme string)`: Extract from Authorization header with optional scheme
+- `FromCookie(key string)`: Extract from HTTP cookies
+- `FromParam(param string)`: Extract from URL path parameters
+- `FromForm(param string)`: Extract from form data
+- `FromHeader(header string)`: Extract from custom HTTP headers
+- `FromQuery(param string)`: Extract from URL query parameters
+- `FromCustom(key string, fn func(fiber.Ctx) (string, error))`: Define custom extraction logic with metadata
+- `Chain(extractors ...Extractor)`: Chain multiple extractors with fallback
+
+### Source Inspection
+
+The `Source` field enables security-aware extraction by identifying the origin of extracted values:
+
+```go
+switch extractor.Source {
+case SourceAuthHeader:
+    // Authorization header - commonly used for authentication tokens
+case SourceHeader:
+    // Custom HTTP headers - application-specific data
+case SourceCookie:
+    // HTTP cookies - client-side stored data
+case SourceQuery:
+    // URL query parameters - visible in URLs and logs (security consideration)
+case SourceForm:
+    // Form data - POST body data
+case SourceParam:
+    // URL path parameters - route-based data
+case SourceCustom:
+    // Custom extraction logic
+}
+```
+
+### Chain Behavior
+
+The `Chain` function implements fallback logic:
+
+- Returns first successful extraction (non-empty value, no error)
+- If all extractors fail, returns the last error encountered or `ErrNotFound`
+- **Skips extractors with `nil` Extract functions** (graceful error handling)
+- Preserves metadata from first extractor for introspection
+- Stores defensive copy for runtime inspection via the `Chain` field
+
+## Security Considerations
+
+### Source Awareness and Custom Extractors
+
+The `Source` field enables security-aware extraction by identifying the origin of extracted values. However, when using `FromCustom`, middleware cannot determine the source of the extracted value, which can compromise security:
```

Comment:
How is specifying a field, a security feature?

Reply:
It’s an introspection feature which middleware like CSRF use for security. Do you have a better wording idea?